packages/react-query/src/QueryClientProvider.tsx

```diff
@@ -3,46 +3,22 @@ import * as React from 'react'
 import type { QueryClient } from '@tanstack/query-core'
 import type { ContextOptions } from './types'
 
-declare global {
-  interface Window {
-    ReactQueryClientContext?: React.Context<QueryClient | undefined>
-  }
-}
-
 export const defaultContext = React.createContext<QueryClient | undefined>(
   undefined,
 )
-const QueryClientSharingContext = React.createContext<boolean>(false)
 
-// If we are given a context, we will use it.
-// Otherwise, if contextSharing is on, we share the first and at least one
-// instance of the context across the window
-// to ensure that if React Query is used across
-// different bundles or microfrontends they will
-// all use the same **instance** of context, regardless
-// of module scoping.
 function getQueryClientContext(
   context: React.Context<QueryClient | undefined> | undefined,
-  contextSharing: boolean,
 ) {
   if (context) {
     return context
   }
-  if (contextSharing && typeof window !== 'undefined') {
-    if (!window.ReactQueryClientContext) {
-      window.ReactQueryClientContext = defaultContext
-    }
-
-    return window.ReactQueryClientContext
-  }
 
   return defaultContext
```

Comment:
What do you think about making it a simple `OR` instead? 
```suggestion
  return context || defaultContext
```

Reply:
Yeah I can do this for the react and solid implementations. If there are any conventions I'm breaking by doing this then I'd love to know from one of the maintainers :)

Reply:
`context ?? defaultContext` is what I'd prefer :)

Reply:
Doesn't `??` transpile to a tone of code adding bundle size?

Reply:
not with the plans we have in:
- #4679

Reply:
Read through it before, but didn't register that part for some reason! Hell yea, stocked!